src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -1415,7 +1418,7 @@ public boolean mayContainAssumingKeyIsInRange(DecoratedKey key)
     protected static final class InstanceTidier implements Tidy
     {
         private final Descriptor descriptor;
-        private final TableId tableId;
+        private final Owner owner;
```

Comment:
Does it make sense for this to be a weak reference? Can the CFS be collected before the sstables?

Reply:
I don't know TBH. But since the owner is passed explicitly when the sstable instance is built, I would find it quite weird that sstable could lost such reference. Especially that the building code doesn't necessarily need to pass an owner, it is allowed to pass null. This is just my gut feeling, and I'm not strongly opposed to change that.